racker/salus-telemetry-model
# Removed MyISAM engine usage in migration 1.1

# Resolves

https://jira.rax.io/browse/SALUS-772

# What

With a newly created managed MySQL in Google Cloud we were seeing migration 1.1 fail with this error:

```

Caused by: org.flywaydb.core.internal.sqlscript.FlywaySqlScriptException:
--
Migration V1_1__hibernate_sequence.sql failed
---------------------------------------------
SQL State  : HY000
Error Code : 3161
Message    : Storage engine MyISAM is disabled (Table creation is disallowed).
Location   : db/migration/mysql/V1_1__hibernate_sequence.sql (/file:/app/libs/salus-telemetry-model-0.1.0-SNAPSHOT.jar!/db/migration/mysql/V1_1__hibernate_sequence.sql)
Line       : 17
Statement  : create table hibernate_sequence
(
next_val bigint not null
)
auto_increment = 1,
engine = MyISAM
 
at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.handleException(DefaultSqlScriptExecutor.java:253)
at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.executeStatement(DefaultSqlScriptExecutor.java:202)
at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.execute(DefaultSqlScriptExecutor.java:125)
at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.execute(SqlMigrationExecutor.java:77)
at org.flywaydb.core.internal.command.DbMigrate.doMigrateGroup(DbMigrate.java:367)
... 38 common frames omitted
Caused by: java.sql.SQLException: Storage engine MyISAM is disabled (Table creation is disallowed).
```

# How

Since the existing clusters were baselined at 1.1, which means no checksum is tracked, we get to cheat and modify this existing migration file to ensure fresh deployments to succeed. Here's what the baseline entry looks like in one of our existing clusters:

```
*************************** 1. row ***************************
installed_rank: 1
       version: 1.1
   description: "hibernate sequence"
          type: BASELINE
        script: "hibernate sequence"
      checksum: NULL
  installed_by: null
  installed_on: 2019-12-05 15:58:13
execution_time: 0
       success: 1
```

For our local MySQL instances you will either need to wipe the volume and let all the migrations re-run on next app startup or replace the checksum value of migration version 1.1 to the value `1236433232`.

## How to test

I partially tested the changed by wiping my mysql volume and letting all the migrations run again. Ideally I should configure mysql to disallow MyISAM engine, but will just save that for perf cluster.
